src/views/ClickupLogin.jsx

```diff
@@ -9,6 +9,7 @@ function ClickupLogin(props) {
   // Fetch OAuth code from redirect URL
   const urlParams = new URLSearchParams(window.location.search)
   const clickupCode = urlParams.get('code')
+  const [performingTokenLogin, setPerformingTokenLogin] = useState(false)
```

Comment:
I added the performingTokenLogin state here to resolve a weird bug Johnny encountered. Essentially, upon attempting a ClickUp login, this component would attempt to log into ClickUp several times. (This didn't occur in the previous version of PA.) This might be because my additions / modifications to a parent component cause the Frame component to refresh multiple times, perhaps because of cookie updates, thus rerunning the ClickUp login code before any of the previous runs have completed. The code here started the login process asynchronously, but didn't check if it was already performing the login. My modification ensures that it only attempts the login once, and waits until that single attempt is finished before reattempting login.

Reply:
Ohh interesting! I wonder if you added an empty array as a dependency to the useEffect, if that would help? I don't know if this is the exact same case, but the react the [documentation](https://react.dev/reference/react/useEffect#reference) says if you omit the dependency array the useEffect will re-run after every re-render of the component. And there doesn't seem to be a dependency array here.

Reply:
Cool!!! I didn't realize the empty array did this. I'll implement this instead of my performingTokenLogin logic :D